external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -2327,36 +2327,123 @@ fn parse_function_decl(
     start_loc: usize,
     modifiers: Modifiers,
     context: &mut Context,
-) -> Result<Function, Box<Diagnostic>> {
+) -> Result<(Function, Option<Box<Diagnostic>>), Box<Diagnostic>> {
     let Modifiers {
         visibility,
         entry,
         native,
         macro_,
     } = modifiers;
 
+    // certain portions of of the partially constructed function must be updated on return
+    fn update_partial(start_loc: usize, context: &mut Context, mut result: Function) -> Function {
+        result.loc = make_loc(
+            context.tokens.file_hash(),
+            start_loc,
+            context.tokens.previous_end_loc(),
+        );
+        result.body = sp(context.tokens.current_token_loc(), FunctionBody_::Native);
```

Comment:
I'm dubious about marking this native, as it will likely produce very strange results. Inserting `abort 0` or something is safer. We could even consider adding a `FunctionBody_::Error` form.

Reply:
This is exactly the kind of feedback I am looking for! I was wondering about it myself (also because we introduce an essentially fake location here), but did not want to go to crazy with the changes without getting feedback first.

In a similar vein (though perhaps more acceptable) is that an incomplete function actually has a `Type_::Unit` return type (rather then perhaps something like `Type_::Error`).

Reply:
+1 For something like `FunctionBody_::Error`
Though you could probably achieve the same thing by doing `FunctionBody_::Defined(...UnresolvedError...)`

Reply:
> +1 For something like `FunctionBody_::Error` Though you could probably achieve the same thing by doing `FunctionBody_::Defined(...UnresolvedError...)`

I went this route (actually just provided empty `Sequence` as the function body) as this does not require changes to subsequent passes (to handle the new type of `UnresolvedError`).